packages/yew/src/dom_bundle/btag/attributes.rs

```diff
@@ -247,3 +274,111 @@ impl Apply for Attributes {
         }
     }
 }
+
+#[cfg(target_arch = "wasm32")]
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use gloo::utils::document;
+    use js_sys::Reflect;
+    use wasm_bindgen_test::{wasm_bindgen_test as test, wasm_bindgen_test_configure};
+
+    use super::*;
+    use crate::{function_component, html, Html};
+
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    fn create_element() -> (Element, BSubtree) {
+        let element = document()
+            .create_element("a")
+            .expect("failed to create element");
+        let btree = BSubtree::create_root(&element);
+        (element, btree)
+    }
+
+    #[test]
+    fn properties_are_set() {
+        let attrs = Attributes::Static(&[
+            ("href", "https://example.com/", ApplyAttributeAs::Property),
+            ("alt", "somewhere", ApplyAttributeAs::Property),
+        ]);
+        let (element, btree) = create_element();
+        attrs.apply(&btree, &element);
+        assert_eq!(
+            Reflect::get(element.as_ref(), &JsValue::from_str("href"))
+                .expect("no href")
+                .as_string()
+                .expect("not a string"),
+            "https://example.com/",
+            "property `href` not set properly"
+        );
+        assert_eq!(
+            Reflect::get(element.as_ref(), &JsValue::from_str("alt"))
+                .expect("no alt")
+                .as_string()
+                .expect("not a string"),
+            "somewhere",
+            "property `alt` not set properly"
+        );
+    }
+
+    #[test]
+    fn respects_apply_as() {
+        let attrs = Attributes::Static(&[
+            ("href", "https://example.com/", ApplyAttributeAs::Attribute),
+            ("alt", "somewhere", ApplyAttributeAs::Property),
+        ]);
+        let (element, btree) = create_element();
+        attrs.apply(&btree, &element);
+        assert_eq!(
+            element.outer_html(),
+            "<a href=\"https://example.com/\"></a>",
+            "should be set as attribute"
+        );
+        assert_eq!(
+            Reflect::get(element.as_ref(), &JsValue::from_str("alt"))
+                .expect("no alt")
+                .as_string()
+                .expect("not a string"),
+            "somewhere",
+            "property `alt` not set properly"
+        );
+    }
+
+    #[test]
+    fn class_is_always_attrs() {
+        let attrs = Attributes::Static(&[("class", "thing", ApplyAttributeAs::Attribute)]);
+
+        let (element, btree) = create_element();
+        attrs.apply(&btree, &element);
+        assert_eq!(element.get_attribute("class").unwrap(), "thing");
+    }
+
+    #[test]
+    async fn macro_syntax_works() {
+        #[function_component]
+        fn Comp() -> Html {
+            html! { <a href="https://example.com/" ~alt="abc" /> }
```

Comment:
I believe in front of the name is better: `~` arguably modifies the name, not the assignment and `~=` is confusing since it can be read as `alt(~=)` instead of `(alt~)=` and the reader is left wondering what the `~=` operator does, is it something like `+=`?

Reply:
That also causes parsing problems since `ident*=expr` is valid Rust but `~` can't be used there so parser doesn't know what to do